shared/zodSchemas.ts

```diff
@@ -15,35 +15,39 @@ export const createGiftSchema = z.object({
   receiver: z.string().min(1),
 });
 
-// CHECK THIS
-// onko alla oleva rivi tarpeellinen?
-export const updateGiftSchema = z.object({
-  gift: z.string().min(1),
-  receiver: z.string().min(1),
-});
+export const updateGiftSchema = createGiftSchema;
 
 export const userSchema = z.object({
+  email: z.string().min(1).max(128).regex(emailRegex).toLowerCase(),
+  firstName: z.string().min(1).max(128),
+  lastName: z.string().min(1).max(128),
+});
+
+export const getUserSchema = userSchema.extend({
   createdAt: z.date(),
   updatedAt: z.date(),
   uuid: z.string(),
-  email: z.string().min(1).max(128).regex(emailRegex),
-  firstName: z.string().min(1).max(128),
-  lastName: z.string().min(1).max(128),
 });
 
-export const createUserSchema = z.object({
-  email: z.string().min(1).max(128).regex(emailRegex),
-  firstName: z.string().min(1).max(128),
-  lastName: z.string().min(1).max(128),
+export const createUserSchema = userSchema.extend({
   password: z.string().min(1).max(128).regex(passwordRegex),
-  Session: z.custom<Prisma.SessionCreateNestedOneWithoutUserInput>().optional(),
 });
 
-export const userLoginDetailsSchema = z.object({
-  email: z.string().min(1).max(100).regex(emailRegex),
-  password: z.string().min(1).max(100).regex(passwordRegex),
+export const updateUserSchema = userSchema;
+
+// CHECK THIS
+export const userLoginDetailsSchema = userSchema
+  .pick({ email: true })
+  .merge(createUserSchema.pick({ password: true }))
+  .extend({ rememberMe: z.boolean() });
+
+/*
+  export const userLoginDetailsSchema = z.object({
+  email: z.string().min(1).max(128).regex(emailRegex),
+  password: z.string().min(1).max(128).regex(passwordRegex),
   rememberMe: z.boolean(),
 });
```

Comment:
Miten tämmöisessä tapauksessa, missä otetaan vain yksi kohta per schema. Kannattaako näissä tehdä kommentoidulla tavalla, vai `merge` ja `extend` tavalla?

Reply:
Kaikki tavat toimii, mutta tuo sun editoitu on noista paras:

```ts
export const userLoginDetailsSchema = createUserSchema
  .pick({ email: true, password: true })
  .extend({ rememberMe: z.boolean() });
```

Yksi vaihtoehto olisi viedä nuo kentät omiin muuttujiin ja käyttää niitä:

```ts
const password = z.string().min(1).max(128).regex(passwordRegex);
const email = z.string().min(1).max(128).regex(emailRegex);

export const createUserSchema = userSchema.extend({
  password:password,

export const userLoginDetailsSchema = userSchema.object({
  email: email,
  password: password,
  rememberMe: z.boolean(),
});
```

Mut ehkä toi sun editoitu on kuitenkin paras.

Reply:
Totta, salasanan ja sähköpostin voisi hyvinkin laittaa omiin kenttiin, erittäin hyvä!! Yritän laittaa korvan taakse tulevaisuutta ajatellen, kiitos! :+1: